enhancements/E-0001-secret-handling.md

```diff
@@ -0,0 +1,748 @@
+
+As a package maintainer I want to be able to request synchronization of a provided secret into Namespaces managed by a ClusterPackage.
+
+### Notes/Constraints/Caveats
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Mutating secrets in place will create problems with workloads not properly watching the filesystem.
+  - These workloads usually use workarounds that annotate their pods with a hash built from the secret object so that its pods are recycled when the secret changes.
+  - Should we take this into consideration?
+  - I was always hoping that we could leverage kustomize's ideas of generating new secrets with new names and referencing the changed secret name instead. This would have given us free workload pod rotation. But I don't think this idea is possible within PKO's model of including plaintext objects in `(Cluster)ObjectDeployments` and `(Cluster)ObjectSets`.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+- Risk of copying the wrong secrets to the wrong places, leaking secrets to an unintended audience.
+  - Although this is a real risk,
+    1. we're not really more afraid of mixing up `Secrets` than mixing up other `Object` types.
+    2. we'll be adding extensive integration tests to the PKO codebase in an attempt to catch any potential bugs which would result in such a leaky scenario
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Changes
+
+#### PackageManifest API changes
+
+The PackageManifest API needs to be changed to include an optional list `.spec.secretRefAliases`, which allows the person writing the object manifests/templates inside the package image to
+  - communicate required secret references to the person deploying the package.
+  - refer to secrets in their templates using the specified alias.
+This effectively decouples package templates and the runtime cluster context and allows the packager to forgo hardcoding magic secret references into the package templates.
+
+
+```yaml
+apiVersion: manifests.package-operator.run/v1alpha1
+kind: PackageManifest
+metadata:
+  name: my-package
+spec:
+  # Optional: specify a list of aliases for referecing secrets in templates.
+  secretRefAliases:
+  - name: tls-keypair-and-certs
```

Comment:
I think we should mandate a description to be provided, so users know what this Secret is used for.

Reply:
I've added an example usage of a `desc` key below. What do you think of it?